app/screens/home/recent_mentions/index.tsx

```diff
@@ -1,24 +1,90 @@
 import {SafeAreaView} from 'react-native-safe-area-context';
+import {of as of$} from 'rxjs';
+import {switchMap} from 'rxjs/operators';
 
-const RecentMentionsScreen = () => {
+import {getRecentMentions} from '@actions/remote/search';
+import DateSeparator from '@app/components/post_list/date_separator';
+import {MM_TABLES, SYSTEM_IDENTIFIERS} from '@app/constants/database';
+import {SystemModel, UserModel} from '@app/database/models/server';
+import {getDateForDateLine, isDateLine, selectOrderedPosts} from '@app/utils/post_list';
+import {getTimezone} from '@app/utils/user';
+import NavigationHeader from '@components/navigation_header';
+import {useServerUrl} from '@context/server_url';
+import {withTheme} from '@context/theme';
+import {useCollapsibleHeader} from '@hooks/header';
+
+import Mention from './components/mention';
+
+import type {WithDatabaseArgs} from '@typings/database/database';
+import type PostModel from '@typings/database/models/servers/post';
+
+const AnimatedFlatList = Animated.createAnimatedComponent(FlatList);
+
+const {USER, SYSTEM, POST} = MM_TABLES.SERVER;
+
+type Props = {
+    currentTimezone: string | null;
+    currentUser: UserModel;
+    isTimezoneEnabled: boolean;
+    mentions: PostModel[];
+    theme: Theme;
+}
+
+const styles = StyleSheet.create({
+    loading: {
+        alignItems: 'center',
+        flex: 1,
+        justifyContent: 'center',
+    },
+});
+
+const RecentMentionsScreen = ({mentions, theme, currentUser, currentTimezone, isTimezoneEnabled}: Props) => {
     const route = useRoute();
     const params = route.params! as {direction: string};
     const toLeft = params.direction === 'left';
-    const [start, setStart] = useState(false);
 
-    const onLayout = useCallback(() => {
-        setStart(true);
-    }, []);
+    const {formatMessage} = useIntl();
+    const [refreshing, setRefreshing] = useState(false);
+    const [loading, setLoading] = useState(false);
+
+    const subtitle = formatMessage({id: 'screen.mentions.subtitle', defaultMessage: 'Messages you\'ve been mentioned in'});
+    const title = formatMessage({id: 'screen.mentions.title', defaultMessage: 'Recent Mentions'});
+    const isLargeTitle = true;
+    const isFocused = useIsFocused();
+    const {scrollPaddingTop, scrollRef, scrollValue, onScroll} = useCollapsibleHeader<FlatList<string>>(isLargeTitle, Boolean(subtitle), false);
+    const paddingTop = useMemo(() => ({paddingTop: scrollPaddingTop, paddingBottom: 10}), [scrollPaddingTop]);
+
+    const serverUrl = useServerUrl();
+
+    useEffect(() => {
+        setLoading(true);
+        getRecentMentions(serverUrl).finally(() => {
+            setLoading(false);
+        });
+    }, [serverUrl]);
+
+    const posts = useMemo(() => selectOrderedPosts(mentions, 0, false, '', false, isTimezoneEnabled, currentTimezone, false).reverse(), [mentions]);
+
+    const handleRefresh = useCallback(async () => {
+        setRefreshing(true);
```

Comment:
If we have stuff in the database, getRecentMentions will not return until the stuff is gotten from the server, right? So the information won't show until we get the information. This will mean that we only take advantage of the stuff being in the database when we are offline. Is this intended?

Reply:
@larkox why do you say that? 
The comment is pointing to `setRefreshing` which is only for the pull down to refresh function!
Is it about the `loading` state? we only show the loader when mentions are empty.

Reply:
Oh! I totally misread the code. I thought that down in the render the condition was only taking in mind whether it was loading or not. Sorry about that.